### PR TITLE
Update CONTRIBUTING docs with a style guide

### PR DESCRIPTION
**Pull request by shaleh**
Add a pitfall section as well.
Add a new CONTRIBUTORS file.

### CODE REVIEW

**Comment by shaleh** [comment]
I would love to be able to generate the CONTRIBUTORS file. This should be possible.

**Comment by shaleh** [comment]
I forgot to reference #412 which this PR would close if landed.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,26 +10,42 @@ Thank you for contributing to Remacs!
...
+  You can learn the exact version we use at any time by looking at the
+  .travis.yml file.
 * Add docstrings to your Rust functions `/// This function does ...`
-* _Really_ great PRs include tests. Don't worry if it's not
-  possible to write tests for your code yet (this is common).
+* _Really_ great PRs include tests. It is not always possible to
+  write unit tests for Rust code if it needs to also call C code. We
+  do have a mocking library available to help with this. See
+  `rust_src/src/strings.rs` for examples. Don't worry if it's not
+  possible to write tests for your code yet (this is common). Another
```
Comment (Wilfred) [inline CONTRIBUTING.md]: Could you split this into two or three paragraphs? It's quite big at the moment.

Comment (shaleh) [reply CONTRIBUTING.md]: I will take another stab at it.

**Pull request merged**